plugins/ruby/ruby.cpp

```diff
@@ -1023,7 +1023,7 @@ static VALUE rb_dfmemory_set_clear(VALUE self, VALUE set)
 
 
 /* call an arbitrary object virtual method */
-#ifdef WIN32
+#if defined(_WIN32) && !defined(_WIN64)
```

Comment:
Microsoft's [documentation](https://msdn.microsoft.com/en-us/library/windows/hardware/ff561499.aspx) shows that calling conventions are effectively identical:

>Unlike the x86, the C/C++ compiler only supports one calling convention on x64. This calling convention takes advantage of the increased number of registers available on x64:
>* The first four integer or pointer parameters are passed in the **rcx**, **rdx**, **r8**, and **r9** registers.
>* The first four floating-point parameters are passed in the first four SSE registers, **xmm0**-**xmm3**.
>* The caller reserves space on the stack for arguments passed in registers. The called function can use this space to spill the contents of registers to the stack.
>* Any additional arguments are passed on the stack.

...

>The calling convention for C++ is very similar: the **this** pointer is passed as an implicit first parameter. The next three parameters are passed in registers, while the rest are passed on the stack.

However this does mean functions expecting floating-point arguments are going to receive garbage data.